# This is a patch for gnucobol-4.0-early-dev to update it to gnucobol-4.0-early-dev-patched
# 
# To apply this patch:
# STEP 1: Chdir to the source directory.
# STEP 2: Run the 'applypatch' program with this patch file as input.
#
# If you do not have 'applypatch', it is part of the 'makepatch' package
# that you can fetch from the Comprehensive Perl Archive Network:
# http://www.perl.com/CPAN/authors/Johan_Vromans/makepatch-x.y.tar.gz
# In the above URL, 'x' should be 2 or higher.
#
# To apply this patch without the use of 'applypatch':
# STEP 1: Chdir to the source directory.
# STEP 2: Run the 'patch' program with this file as input.
#
#### End of Preamble ####

#### Patch data follows ####
diff -Naud 'gnucobol-4.0-early-dev/cobc/tree.c' 'gnucobol-4.0-early-dev-patched/cobc/tree.c'
Index: ./cobc/tree.c
--- ./cobc/tree.c	Thu Jul 29 08:58:00 2021
+++ ./cobc/tree.c	Mon Aug  9 22:03:25 2021
@@ -206,8 +206,6 @@
 static	int	prev_expr_warn[EXPR_WARN_PER_LINE] = {0,0,0,0,0,0,0,0};
 static	int	prev_expr_tf[EXPR_WARN_PER_LINE] = {0,0,0,0,0,0,0,0};
 
-static struct cb_report *report_checked = NULL;
-
 /* Local functions */
 
 static int
@@ -3990,8 +3988,8 @@
 	struct cb_reference	*ref;
 	int		k;
 
-	if (report_checked != r) {
-		report_checked = r;
+	if (!r->checked) {
+		r->checked = 1;
 		if (r->lines > 9999) {
 			r->lines = 9999;
 		}
diff -Naud 'gnucobol-4.0-early-dev/cobc/tree.h' 'gnucobol-4.0-early-dev-patched/cobc/tree.h'
Index: ./cobc/tree.h
--- ./cobc/tree.h	Thu Jul 29 08:58:00 2021
+++ ./cobc/tree.h	Mon Aug  9 22:03:52 2021
@@ -1537,6 +1537,7 @@
 	unsigned int		global:1;	/* IS GLOBAL declared */
 	unsigned int		has_declarative:1;/* Has Declaratives Code to be executed */
 	unsigned int		has_detail:1;	/* Has DETAIL line */
+	unsigned int            checked:1;
 };
 
 #define CB_REPORT(x)	(CB_TREE_CAST (CB_TAG_REPORT, struct cb_report, x))
#### End of Patch data ####

#### ApplyPatch data follows ####
# Data version        : 1.0
# Date generated      : Mon Aug  9 22:05:02 2021
# Generated by        : makepatch 2.03
# Recurse directories : Yes
# Excluded files      : (\A|/).*\~\Z
#                       (\A|/).*\.a\Z
#                       (\A|/).*\.bak\Z
#                       (\A|/).*\.BAK\Z
#                       (\A|/).*\.elc\Z
#                       (\A|/).*\.exe\Z
#                       (\A|/).*\.gz\Z
#                       (\A|/).*\.ln\Z
#                       (\A|/).*\.o\Z
#                       (\A|/).*\.obj\Z
#                       (\A|/).*\.olb\Z
#                       (\A|/).*\.old\Z
#                       (\A|/).*\.orig\Z
#                       (\A|/).*\.rej\Z
#                       (\A|/).*\.so\Z
#                       (\A|/).*\.Z\Z
#                       (\A|/)\.del\-.*\Z
#                       (\A|/)\.make\.state\Z
#                       (\A|/)\.nse_depinfo\Z
#                       (\A|/)core\Z
#                       (\A|/)tags\Z
#                       (\A|/)TAGS\Z
# p 'cobc/tree.c' 170804 1628568205 0100644
# p 'cobc/tree.h' 85913 1628568232 0100644
#### End of ApplyPatch data ####

#### End of Patch kit [created: Mon Aug  9 22:05:02 2021] ####
#### Patch checksum: 71 2463 27706 ####
#### Checksum: 89 3176 21687 ####
